src/main/java/frc/robot/Robot.java

```diff
@@ -15,89 +47,157 @@
-    // this line or comment it out.
-    if (m_autonomousCommand != null) {
-      m_autonomousCommand.cancel();
+
+
+    private void configureBindings() {
+        // Driver
+        oi.getDriverTrigger(Controls.INTAKE).onTrue(transitionState(RobotState.INTAKE));
+        oi.getDriverButton(Controls.REVERSE).onTrue(transitionState(RobotState.SUCK_IN).andThen(transitionState(RobotState.IDLE)));
+        oi.getDriverButton(Controls.AMP).onTrue(transitionState(RobotState.AMP));
+        oi.getDriverTrigger(Controls.FENDER).onTrue(transitionState(RobotState.FENDER));
+        oi.getDriverButton(Controls.REVERSE).onTrue(transitionState(RobotState.REVERSE)).onFalse(transitionState(RobotState.IDLE));
+
+        oi.getDriverButton(Controls.IDLE).onTrue(transitionState(RobotState.IDLE));
+
+
+        // Operator
+        oi.getOperatorButton(Controls.CLIMB_UP).onTrue(climb.moveUp()).onFalse(climb.stop());
+        oi.getOperatorButton(Controls.CLIMB_DOWN).onTrue(climb.moveDown()).onFalse(climb.stop());
+        oi.getOperatorButton(Controls.OPERATOR_REVERSE).onTrue(transitionState(RobotState.SUCK_IN));
+
+        oi.getOperatorButton(Controls.ZERO_GYRO).onTrue(swerve.zeroGyro());
+        oi.getOperatorButton(Controls.ZERO_ENCODERS).onTrue(new InstantCommand(() -> swerve.updateEncoders()));
+
+        oi.getOperatorButton(Controls.IDLE).onTrue(transitionState(RobotState.IDLE));
+
+        oi.getOperatorButton(XboxController.Button.kLeftBumper.value).onTrue(shooter.toggaleVoltage());
+
+        // oi.getDriverController().b().onTrue(new WheelRadiusChar(swerve, Constants.Swerve.MODULE_TYPE, Constants.Swerve.DRIVE_BASE_RADIUS));
+
+        swerve.setDefaultCommand(new TeleopSwerveDrive(swerve, oi));
+    }
+
+
+    public Command transitionState(RobotState newState) {
+        return new SequentialCommandGroup (
+            new InstantCommand(() -> {
+                currState = newState;
+                System.out.println("Transitining to state: " + newState.toString() + " " + Timer.getMatchTime());
+            }),
+            wrist.changeState(newState),
+            new ParallelCommandGroup(
+                intake.changeState(newState),
+                shooter.changeState(newState)
+            )
+        );
+    }
+
+    @Override
+    public void robotPeriodic() {
+        CommandScheduler.getInstance().run();
+
+        SmartDashboard.putString("Current State", currState.toString());
+
+        //if(intake.hasNote()){
+        //    enableRumble().schedule();
+        //    disableRumble().schedule();
+        //}
+    }
+
+    @Override
+    public void disabledInit() {}
+
+    @Override
+    public void disabledPeriodic() {
+        Swerve.getInstance().updateEncoders();
+    }
+
+    @Override
+    public void autonomousInit() {
+        m_autonomousCommand = getAutonomousCommand();
+
+        if (m_autonomousCommand != null) {
+            System.out.println("Auton starting " + Timer.getMatchTime());
```

Comment:
not sure why were logging this? match time is always zero when starting auto + already logged in ds